### PR TITLE
Defaults module

### LINKED ISSUE #52: Defaults module
[Spun-off from discussion in #51.]

Create a `defaults.py` that defines defaults as constants. These are then used in `train.py` and `predict.py`, and in some cases, in the architecture modules themselves.

### PR DESCRIPTION
**Pull request by kylebgorman**
Adds a module for default constants and uses them.

This closes #52.

### CODE REVIEW

File: yoyodyne/collators.py [hunk 1]
```diff
@@ -28,8 +27,8 @@ def __init__(
         pad_idx,
         config: dataconfig.DataConfig,
         arch: str,
-        max_source_length: int = 128,
-        max_target_length: int = 128,
+        max_source_length=defaults.MAX_SOURCE_LENGTH,
```
Comment (Adamits) [inline yoyodyne/collators.py:30]: Is removing the type annotation here intentional?

**Comment by Adamits** [comment]
Thanks for doing this, and for adding the collators argparse stuff!

I did not check exhaustively if any args were missed, but this LGTM.

**Review by Adamits — APPROVED** [review]
Thanks for doing this, and for adding the collators argparse stuff!

I did not check exhaustively if any args were missed, but this LGTM.

File: yoyodyne/collators.py (hunk 1 above)
Comment (kylebgorman) [reply yoyodyne/collators.py]: Reverted.

**Pull request merged**